src/lib/core/diag.h

```diff
@@ -53,6 +53,7 @@ struct type_info;
 struct error;
 
 typedef void (*error_f)(struct error *e);
+typedef struct error *(*error_dup_f)(const struct error *e);
```

Comment:
Even though it's consistent with other function types defined in this file, I still think we should follow the checkpatch recommendation, because it's more common in Tarantool code:

```C
typedef struct error *
(*error_dup_f)(const struct error *e);
```

Reply:
Done.